core/src/main/scala/org/apache/spark/scheduler/CoarseCookSchedulerBackend.scala

```diff
@@ -349,80 +373,76 @@ class CoarseCookSchedulerBackend(
         }
       }
     }
-  }
 
-  /**
-   * Kill the given list of executors through the cluster manager.
-   *
-   * @return whether the kill request is acknowledged.
-   */
   override def doKillExecutors(executorIds: Seq[String]): Future[Boolean] =
     Future.successful {
-      val jobIdsToKill = executorIds.flatMap(executorIdToJobId.get)
-      jobClient.abort(jobIdsToKill.asJava)
-      jobIdsToKill.foreach(abortedJobIds.add)
-      true
+      // No need to adjust `executorLimitOption` since the AllocationManager already communicated
+      // the desired limit through a call to `doRequestTotalExecutors`.
+      // See [[o.a.s.scheduler.cluster.CoarseGrainedSchedulerBackend.killExecutors]]
+      val jobUUIDsToAbort = executorIds.flatMap(executorIdToJobUUID.get).toList
```

Comment:
I am 100% sure that won't happen but it won't hurt to make the code more defensive. Will change